src/hooks/Tokens.ts

```diff
@@ -117,14 +118,32 @@ export interface CoingeckoData {
  * */
 
 export function useCoinGeckoTokenData(coin: Token) {
-  const chain = CHAINS[coin.chainId].mainnet ? CHAINS[coin.chainId] : CHAINS[ChainId.AVALANCHE]
+  const chain = useChain(coin.chainId)
 
   return useQuery(['coingeckoToken', coin.address, chain.name], async () => {
-    if (!chain.coingecko_id) {
+    if (!chain.coingecko_id || !chain.evm) {
       return null
     }
-    const response = await fetch(`${COINGECKO_API}/coins/${chain.coingecko_id}/contract/${coin.address.toLowerCase()}`)
-    const data = await response.json()
+    const response = await axios
+      .get(`${COINGECKO_API}/coins/${chain.coingecko_id}/contract/${coin.address.toLowerCase()}`, {
+        timeout: 1000
```

Comment:
timeout should be bigger otherwise this api call will fail if internet connection is slow.